### DISCUSSION TITLE
Is it possible to call custom keywords in the hook?

### DISCUSSION THREAD

**Discussion by Heus-Sueh** [InioX/matugen#93] [Q&A]
I ask this because apparently the script is not being called

custom keywords
```
[config.custom_keywords]
hooks = "$HOME/.config/matugen/hooks"
```

template and hook
```
[templates.bat]
input_path = '~/.config/matugen/templates/bat-colors.tmTheme'
output_path = '~/.config/bat/themes/matugen.tmTheme'
hook = "./{{custom.hooks}}/bat-hook.sh"
```

script
```
#!/bin/bash
echo "Changing the colors of bat..."
bat cache --build
echo "Changed the colors of bat successfully!"

```

**Comment by InioX** [comment]
Try using an absolute path instead of `$HOME`.

**Reply by Heus-Sueh** [reply]
I discovered the problem, the `hook` function is interconnected with the `colors_to_compare` function, so you can't just use `hook` without also calling the `colors_to_compare` and `compare_to` functions

[off-topic]
Could you add variables to `input_path`?

I wanted to do something like this:

```
starship = { input_path = '{{custom.templates}}/starship-colors.toml', output_path = '~/.config/starship.toml' }

```

**Reply by InioX** [reply]
I honestly have no idea why I decided that hooks should only execute if `colors_to_compare` is defined. I'll fix it and add variables to `input_path` as well.

**Reply by InioX** [reply]
Should work now.
